### PR TITLE
Populate OpenAPI in all integration tests

### PR DESCRIPTION
**Pull request by Jefftree**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

SSA is enabled on an as needed basis in integration tests by populating the OpenAPI field of the configuration. APF recently switched to use SSA in their controller, and SSA is now needed in all integration tests. This PR populates the OpenAPI in all integration tests.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by liggitt** [comment]
if we always want/expect FieldManager to be set, should this part of the server setup become unconditional and stop tolerating nil `OpenAPIModels`?

```
	if a.group.OpenAPIModels != nil && utilfeature.DefaultFeatureGate.Enabled(features.ServerSideApply) {
```

File: test/integration/framework/controlplane_utils.go [hunk 1]
```diff
@@ -285,6 +285,9 @@ func NewIntegrationTestControlPlaneConfigWithOptions(opts *ControlPlaneConfigOpt
 	// TODO: get rid of these tests or port them to secure serving
 	controlPlaneConfig.GenericConfig.SecureServing = &genericapiserver.SecureServingInfo{Listener: fakeLocalhost443Listener{}}
 
+	// Enable OpenAPI and Server Side Apply in integration tests
+	controlPlaneConfig.GenericConfig.OpenAPIConfig = DefaultOpenAPIConfig()
```
Comment (liggitt) [inline test/integration/framework/controlplane_utils.go:289]: only set if nil?

Comment (Jefftree) [reply test/integration/framework/controlplane_utils.go]: Updated

**Comment by liggitt** [comment]
/lgtm

did you check the integration test log to make sure no other server setup paths needed adjusting?

**Comment by Jefftree** [comment]
> if we always want/expect FieldManager to be set, should this part of the server setup become unconditional and stop tolerating nil `OpenAPIModels`?
> 
> ```
> 	if a.group.OpenAPIModels != nil && utilfeature.DefaultFeatureGate.Enabled(features.ServerSideApply) {
> ```

I see @apelisse has already given the thumbs up. My instinct is also yes but I'd like to do a bit more investigation around that.

**Comment by liggitt** [comment]
> I see @apelisse has already given the thumbs up. My instinct is also yes but I'd like to do a bit more investigation around that.

fine to do in a follow up

**Comment by apelisse** [comment]
Another thing we have is that we enable the server-side apply feature in a lot of the ssa tests, should we remove these calls?

**Comment by Jefftree** [comment]
SSA was GA in 1.22. Is there a minimum number of releases we need to wait before removing the feature flag?

**Comment by apelisse** [comment]
Well, I'm not even mentioning removing the feature flag (I've seen discussion that might answer this question, but I'm not sure, I think it possibly breaks the command-line), but the tests can probably now assume that the feature is enabled?

**Comment by liggitt** [comment]
the integration test job started timing out dramatically last night, trying to figure out if this is related: 
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-integration

**Comment by MikeSpreitzer** [comment]
@MikeSpreitzer

**Pull request merged**